src/main/java/transforms/DateTimeTransform.java

```diff
@@ -25,9 +26,21 @@ public DateTimeTransform(
             @JsonProperty( Constants.TIMEZONE ) Optional<String> timezone
     ) {
         this.pattern = pattern;
-        if (timezone.isPresent()){
-            this.timezone = TimeZone.getTimeZone( timezone.get() );
+
+        if ( timezone.isPresent() ) {
+            String timezoneId = timezone.get();
+
+            if ( !TimeZones.supportedZoneIds.contains( timezoneId ) ) {
```

Comment:
A much more portable check here !TimeZone.getTimeZone( timezoneId ).getId().equals( timezoneId ). i.e if parsed timezoneId doesn't match provided timezoneId then invalid timezoneId was specified and it default to GMT.

Reply:
oooh true!